feat: add --kernel flag to run command

avorylli opened the pull request:
Fixes #1321

Adds `--kernel` flag to the `run` command that allows specifying a kernel assembly file (.masm) to compile and configure for the VM.

Changes:
- Add `--kernel` flag to `RunCmd` struct
- Implement kernel compilation from assembly file
- Use compiled kernel when assembling the program
- Add `ast()` method to `ProgramFile` for AST access
- Update all `get_masm_program` call sites for backward compatibility

plafer commented on miden-vm/src/cli/debug/mod.rs:
```diff
@@ -57,7 +57,7 @@ impl DebugCmd {
                 (get_masp_program(&self.program_file)?, Arc::new(DefaultSourceManager::default()))
             },
-            "masm" => get_masm_program(&self.program_file, &libraries, true)?,
+            "masm" => get_masm_program(&self.program_file, &libraries, true, None)?,
```
let's add a `--kernel` flag to `miden debug`, `miden prove`, and `miden verify` as well.

For `miden verify` [here](https://github.com/0xMiden/miden-vm/blob/f3a52ca8535df570f081b5ed2c8f61de8d184753/miden-vm/src/cli/verify.rs#L53-L54) we'll need to load the kernel, and just extract `KernelLibrary.kernel`.

avorylli commented:
> Hey @avorylli the CI still fails, could you
> 
> 1. merge the latest `next`
> 2. run `make clippy` and fix all errors
> 3. make sure tests pass with `make test`
> 4. format with `cargo +nightly fmt`
> 5. add a changelog entry
> 
> This should make the CI green, after which we can merge

Thank you for guiding me! 
- so I merged "next", 
- ran clippy and fixed them, 
- ran make tests and there were some errors but I think they were not dependant on my changes, 
- formatted with fmt
- added a changelog

if anything's wrong please tell me, I'll be happy to correct everything so it matches Miden standards
